hallcristobal/tpgz
# Prevent accidental cursor movement when opening menu

### Intention
It is easy to move the cursor down by mistake when opening the menu. This is because dpadDown is used in opening the menu while also being used to move the cursor down. The result is that if a user does not consciously try to press and release dpadDown quickly when opening the menu, the cursor ends up moving. Obviously this is not user-friendly, so this PR seeks to solve the issue.

### Implementation Details
When the menu first opens, the user is holding L+R+dpadDown. We have to ignore the dpadDown here or else the cursor would move. We could wait for a frame where there is no dpadDown input, then accept a dpadDown input on any frames after that point potentially. But this does not work because the inputs which open the menu have a tendency to flicker on and off right after the menu opens (tested on 2 different controllers and was seeing this consistently).

So basically there is a short buffer which accounts for any sort of input flickering, and it is also used to prevent accidental cursor movement from things like the user releasing L+R a frame before releasing dpadDown.

### Testing
For testing, I recommend the following procedure:
- Create an iso from the latest on master (naming it something like tpgz_master).
- Create an iso from this branch (naming it something like tpgz_cursorPr).
- Observe the current behavior in the tpgz_master iso:
  - Open the menu by **comfortably** pressing L+R+dpadDown (making sure to not execute and release the inputs unreasonably fast to account for the current behavior we are trying to fix). The cursor will likely move down one or two positions.
- Next, run the tpgz_cursorPr iso and open the menu in the same way. The cursor should not move by mistake when opening the menu.

### Final Notes
I pressed L+R+dpadDown quite a large number of times while refining the logic so I am confident in it as it stands, but I would be happy to answer any questions there might be.
